src/com/facebook/buck/cli/TestRunning.java

```diff
@@ -788,41 +788,50 @@ private static Step getReportCommand(
     Set<String> srcFolders = Sets.newHashSet();
     loopThroughSourcePath:
     for (Path javaSrcPath : javaSrcs) {
-      if (!MorePaths.isGeneratedFile(javaSrcPath)) {
-        // If the source path is already under a known source folder, then we can skip this
-        // source path.
-        for (String srcFolder : srcFolders) {
-          if (javaSrcPath.startsWith(srcFolder)) {
-            continue loopThroughSourcePath;
-          }
-        }
+      if (MorePaths.isGeneratedFile(javaSrcPath)) {
+        continue;
+      }
 
-        // If the source path is under one of the source roots, then we can just add the source
-        // root.
-        ImmutableSortedSet<String> pathsFromRoot = defaultJavaPackageFinder.getPathsFromRoot();
-        for (String root : pathsFromRoot) {
-          if (javaSrcPath.startsWith(root)) {
-            srcFolders.add(root);
-            continue loopThroughSourcePath;
-          }
+      // If the source path is already under a known source folder, then we can skip this
+      // source path.
+      for (String srcFolder : srcFolders) {
+        if (javaSrcPath.startsWith(srcFolder)) {
+          continue loopThroughSourcePath;
         }
+      }
 
-        // Traverse the file system from the parent directory of the java file until we hit the
-        // parent of the src root directory.
-        ImmutableSet<String> pathElements = defaultJavaPackageFinder.getPathElements();
-        Path directory = filesystem.getPathForRelativePath(javaSrcPath.getParent());
-        while (directory != null && !pathElements.contains(directory.getFileName().toString())) {
-          directory = directory.getParent();
+      // If the source path is under one of the source roots, then we can just add the source
+      // root.
+      ImmutableSortedSet<String> pathsFromRoot = defaultJavaPackageFinder.getPathsFromRoot();
+      for (String root : pathsFromRoot) {
+        if (javaSrcPath.startsWith(root)) {
+          srcFolders.add(root);
+          continue loopThroughSourcePath;
         }
+      }
 
-        if (directory != null) {
-          String directoryPath = directory.toString();
-          if (!directoryPath.endsWith("/")) {
-            directoryPath += "/";
-          }
-          srcFolders.add(directoryPath);
-        }
+      // Traverse the file system from the parent directory of the java file until we hit the
+      // parent of the src root directory.
+      ImmutableSet<String> pathElements = defaultJavaPackageFinder.getPathElements();
+      Path directory = filesystem.getPathForRelativePath(javaSrcPath.getParent());
+      if (pathElements.isEmpty()) {
+        continue;
+      }
+
+      while (directory != null && directory.getFileName() != null &&
+          !pathElements.contains(directory.getFileName().toString())) {
+        directory = directory.getParent();
+      }
+
+      if (directory == null && directory.getFileName() == null) {
```

Comment:
This is a compilation error since `directory` must be `null` when you reference it the second time (after the `&&`)